### PR TITLE
WIP: Feature screenshot

### PR DESCRIPTION
**Pull request by rbachleitner**
The next question is where we want to call it

### CODE REVIEW

**Comment by rbachleitner** [comment]
A thumbnail should now be automatically taken for the pubmed case. The problem with the dotdotdot javascript lib persists. For now it automatically generates the thumbnail with dotdotdot turned off, this shouldnt change anything for regular headstart use.

File: examples/search_repos/headstart.php [hunk 1]
```diff
@@ -13,7 +13,8 @@
         	data_config.files = [{
         		title: <?php echo json_encode($_GET['title']) ?>,
         		file: <?php echo json_encode($_GET['file']) ?>
-        	}]
+        	}];
+        	data_config.nodot = <?php echo json_encode($_GET['nodot']) ?>;
```
Comment (rbachleitner) [inline examples/search_repos/headstart.php]: So I can check in headstart if dotdotdot should truncate the area titles

File: server/preprocessing/conf/config.ini [hunk 2]
```diff
@@ -3,15 +3,18 @@
...
 # Host of the client visualization
-host = "localhost/"
+host = "http://localhost/"
 # Relative path to the client visualization. Needs to be in the public_html/www directory.
-vis_path = "headstart3/"
+vis_path = "examples/search_repos/"
 # Relative path to the client REST services. Needs to be in the public_html/www directory.
-services_path = "headstart3/server/services/"
+services_path = "server/services/"
+phantomjs_path = "/home/rbachleitner/Downloads/phantomjs-2.1.1-linux-x86_64/bin/phantomjs"
```
Comment (pkraker) [inline server/preprocessing/conf/config.ini]: Please create an own section "snapshot" for these settings and provide a description for them.

File: server/services/searchPubmed.php [hunk 3]
```diff
@@ -3,16 +3,28 @@
...
 $dirty_query = library\CommUtils::getParameter($_POST, "q");
 
 $post_params = $_POST;
 
 $result = search("pubmed", $dirty_query, $post_params, array("article_types", "from", "to"), ";", null);
 
+$id = json_decode($result)->id;
+$query = json_decode($result)->query;
+$title = 'Overview of PubMed articles for ' . $query . ':' . $id;
+$command = 'nohup '. $settings["phantomjs_path"] . ' ' . $settings["getsvg_path"] . ' "' . $settings["host"] . 'examples/search_repos/headstart.php?nodot=1&title='.$title.'&service=pubmed' . $post_params['service'] . '&file=' . $id . '" '. $settings["storage_path"] . $id .'.png "1200px" 1>/dev/null 2>/dev/null &';
```
Comment (pkraker) [inline server/services/searchPubmed.php]: Command does not work in Windows. The PHP file that is being used for the conversion should be configurable.

File: server/services/searchPubmed.php [hunk 4]
```diff
@@ -3,16 +3,28 @@
...
+$settings = $ini_array["general"];
+
 $dirty_query = library\CommUtils::getParameter($_POST, "q");
 
 $post_params = $_POST;
 
 $result = search("pubmed", $dirty_query, $post_params, array("article_types", "from", "to"), ";", null);
 
+$id = json_decode($result)->id;
+$query = json_decode($result)->query;
```
Comment (pkraker) [inline server/services/searchPubmed.php]: No need to decode the same result twice.

File: server/services/searchPubmed.php [hunk 5]
```diff
@@ -3,16 +3,28 @@
...
+$title = 'Overview of PubMed articles for ' . $query . ':' . $id;
```
Comment (pkraker) [inline server/services/searchPubmed.php]: The title should be created by the PHP file used for displaying the map.

File: server/services/searchPubmed.php (hunk 3 above)
Comment (rbachleitner) [inline server/services/searchPubmed.php]: Yes there's Linux specific instructions in the command. Will try to change that.

File: examples/search_repos/headstart.php (hunk 1 above)
Comment (rbachleitner) [inline examples/search_repos/headstart.php]: Added a new headstartsnapshot.php, seems easier in !dac94ed

File: server/preprocessing/conf/config.ini [hunk 6]
```diff
@@ -3,15 +3,18 @@
 
 [general]
 # Full path to the preprocessing directory
-preprocessing_dir = "/home/rbachleitner/workspace/ausleihhistory/kc_frontend/vis/js/headstart/server/preprocessing/"
+preprocessing_dir = "/home/rbachleitner/workspace/Headstart_new/server/preprocessing/"
```
Comment (rbachleitner) [inline server/preprocessing/conf/config.ini]: Fixed

File: server/preprocessing/conf/config.ini (hunk 2 above)
Comment (rbachleitner) [inline server/preprocessing/conf/config.ini]: Done

File: server/services/searchPubmed.php (hunk 3 above)
Comment (rbachleitner) [inline server/services/searchPubmed.php]: Will have to look up how php exec() works on windows

File: server/services/searchPubmed.php (hunk 4 above)
Comment (rbachleitner) [inline server/services/searchPubmed.php]: fixed

File: server/services/searchPubmed.php (hunk 5 above)
Comment (rbachleitner) [inline server/services/searchPubmed.php]: Well for search_repos the title is created in search.js and also passed along to headstart.php. Should I change that somehow?

Comment (pkraker) [inline server/services/searchPubmed.php]: Yes, please do. You can take a look at the vis.php in project-website: https://github.com/OpenKnowledgeMaps/project-website/blob/master/vis.php

File: server/services/searchPubmed.php (hunk 3 above)
Comment (pkraker) [inline server/services/searchPubmed.php]: Pretty much the same, but it does not take Linux commands obv. See here: https://github.com/OpenKnowledgeMaps/Headstart/blob/master/server/classes/headstart/preprocessing/calculation/RCalculation.php

**Pull request merged**